src/System.Private.ServiceModel/tests/Common/Infrastructure/src/testproperties.targets

```diff
@@ -1,115 +1,85 @@
-        sb.AppendLine(
-@"        static partial void Initialize(Dictionary<string, string> properties)
-        {");
-                
-        foreach (var item in TestPropertyItems)
-        {
-            string propertyName = item.ItemSpec;
-            string propertyValue = item.GetMetadata("Value");
-            if (String.IsNullOrEmpty(propertyValue))
-            {
-                propertyValue = item.GetMetadata("DefaultValue");
-            }
-         
-            // Convert the property value into a legal string literal.     
-            StringBuilder literal = new StringBuilder();       
-            foreach (var c in propertyValue)
-            {
-                switch (c)
-                {
-                    case '\'': literal.Append(@"\'"); break;
-                    case '\"': literal.Append("\\\""); break;
-                    case '\\': literal.Append(@"\\"); break;
-                    default:
-                        if (c >= 0x20 && c <= 0x7e)
-                        {
-                            // ASCII printable character
-                            literal.Append(c);
-                        }
-                        else
-                        {
-                            // As UTF16 escaped character
-                            literal.AppendFormat(@"\u{0:x4}", (int)c);
-                        }
-                        break;
-                }
-            }
-            
-            string setStatement = String.Format("            properties[\"{0}\"] = \"{1}\";", propertyName, literal.ToString());
-            Log.LogMessage(setStatement);
-            sb.AppendLine(setStatement);
-        }
-            
-	      sb.AppendLine(
- @"        }
     }
-}");
-        
-        // Do not rewrite file if there are no changes.  This permits incremental builds.
-        string contentsToWrite = sb.ToString();
-        string priorContents = File.Exists(FileName) ? File.ReadAllText(FileName) : String.Empty;
-        if (!String.Equals(contentsToWrite, priorContents))
-        {
-            Log.LogMessage("Test properties were generated to {0}.", FileName);
-            File.WriteAllText(FileName, sb.ToString());
-        }
-        else
-        {
-             Log.LogMessage("No changes were made to the existing file {0}", FileName);
-        }
-        ]]>
-      </Code>
-    </Task>
-  </UsingTask>
+}
+                ]]>
+    </_TestPropertyCode>
+  </PropertyGroup>
+
+  <!-- 
+    SanitizeTestProperties:
+      This target ensures the test property values are sanitized for proper
+      evaluation at runtime.
+  -->
+  <Target Name="SanitizeTestProperties">
+    <!-- BridgeResourceFolder needs to convert forward slashes to backward
+         slashes to be usable on Windows, where the Bridge runs.
+    -->
+    <PropertyGroup Condition="$(BridgeResourceFolder.Contains('/'))">
+      <BridgeResourceFolder>$(BridgeResourceFolder.Replace('/', '\'))</BridgeResourceFolder>
```

Comment:
Is this strictly necessary? I've been able to run things on the command line using forward slashes, but I agree it's better to sanitize rather than not.